### PR TITLE
Prow pubsub: allow restricting clusters for project

### PR DESCRIPTION
**Pull request by chaodaiG**
Added a new prow config field while still keeps backward compatible

### CODE REVIEW

File: prow/pubsub/subscriber/subscriber.go [hunk 1]
```diff
@@ -238,6 +238,22 @@ func (s *Subscriber) handleProwJob(l *logrus.Entry, jh jobHandler, msg messageIn
...
+	// deny job that runs on not allowed cluster
+	var clusterIsAllowed bool
+	for _, allowedCluster := range allowedClusters {
+		if allowedCluster == "*" || allowedCluster == prowJobSpec.Cluster {
+			clusterIsAllowed = true
+			break
+		}
+	}
+	if !clusterIsAllowed {
+		err := fmt.Errorf("cluster %s is not allowed", prowJobSpec.Cluster)
```
Comment (cjwagner) [inline prow/pubsub/subscriber/subscriber.go:250]: This isn't very actionable. Please include some brief instructions for adding the necessary config.

**Review by cjwagner — APPROVED** [review]
One nit.
/hold

File: prow/pubsub/subscriber/subscriber.go (hunk 1 above)
Comment (chaodaiG) [reply prow/pubsub/subscriber/subscriber.go]: done

**Review by cjwagner — APPROVED** [review]

**Pull request merged**